internal/provider/resources/deployment.go

```diff
@@ -646,3 +849,47 @@ func (r *DeploymentResource) ImportState(ctx context.Context, req resource.Impor
 		resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("workspace_id"), workspaceID.String())...)
 	}
 }
+
+// validatorsForConflictingAttributes provides a list of string validators
+// used in a ConflictsWith validator for a specific attribute.
+//
+// This approach is used in lieu of a ConfigValidators method because we take
+// advantage of 'MatchRelative' to use the current context of the list objects
+// (ListNestedAttribute).
+//
+// Also, expressing validators on each attribute lets us
+// be more concise when defining the conflicting attributes. Defining them in
+// ConfigValidators instead would be much more verbose, and disconnected from
+// the source of truth.
+func validatorsForConflictingAttributes(attributes []string) []validator.String {
```

Comment:
Configuring conflicting attributes results in an error like this:

```
        Error: Invalid Attribute Combination

          with prefect_deployment.terraformacc4ms7axesye,
          on terraform_plugin_test.tf line 45, in resource "prefect_deployment" "terraformacc4ms7axesye":
          45: resource "prefect_deployment" "terraformacc4ms7axesye" {

        Attribute "pull_steps[0].repository" cannot be specified when
        "pull_steps[0].directory" is specified
```

Reply:
yeah, i think this approach makes sense because you're needing to validate each list member without knowing how many members there will be - when i used the ConfigValidator method, i could specify a deterministic path to an object key, vs. having to account for n-members in something like PullSteps